index.bs

```diff
@@ -503,7 +505,102 @@ Add the step:
+
+<dl dfn-for="pending event">
+: <dfn>attributionReportingEnabled</dfn>
+:: A [=boolean=]
+: <dfn>contextOrigin</dfn>
+:: An [=origin=]
+
+</dl>
+
+Modify {{Fence/reportEvent()}} as follows:
+
+Replace the step
+
+> Run <a spec=fenced-frame>report an event</a> using <var ignore>instance</var>'s
+> [=fenced frame config instance/fenced frame reporter=] with |destination|,
+> |event|'s {{FenceEvent/eventType}}, and |event|'s {{FenceEvent/eventData}}.
+
+with
+
+Run <a spec=fenced-frame>report an event</a> using <var ignore>instance</var>'s
+[=fenced frame config instance/fenced frame reporter=] with |destination|,
+|event|'s {{FenceEvent/eventType}}, |event|'s {{FenceEvent/eventData}},
+and [=this=]'s [=relevant global object=]'s [=associated Document=].
+
+Modify <a spec=fenced-frame>report an event</a> as follows:
+
+Add a [=Document=] parameter called |document|.
+
+After the step
+
+> If <var ignore>reporting map</var>[|destination|] does not [=map/exist=], return.
+
+add the steps
+
+<div algorithm="report-an-event-monkeypatch">
+1. Let <var ignore>attributionReportingEnabled</var> be the result of determining whether the
+    "`attribution-reporting`" feature [=is feature enabled|is enabled=] in |document| with |document|'s [=origin=].
+1. Let <var ignore>contextOrigin</var> be |document|'s [=node/context origin=].
+
+</div>
+
+In the step
+
+> Let <var ignore>newEvent</var> be a new [=fencedframetype/pending event=] with the following...
+
+add the properties
+
+: [=pending event/attributionReportingEnabled=]
+:: |attributionReportingEnabled|
+: [=pending event/contextOrigin=]
+:: |contextOrigin|
+
+Replace the step
+
+> <a spec=fenced-frame>Send a beacon</a> with <var ignore>reporting map</var>[|destination|],
+> |eventType|, and |eventData|.
+
+with
+
+<a spec=fenced-frame>Send a beacon</a> with <var ignore>reporting map</var>[|destination|],
+|eventType|, |eventData|, |attributionReportingEnabled|, and |contextOrigin|.
+
+Modify [=finalize a reporting destination=] as follows:
+
+Replace the step
+
+> <a spec=fenced-frame>Send a beacon</a> with |destination map|, |pending event|'s [=pending event/eventType=],
+> and |pending event|'s [=pending event/eventData=].
+
+with
+
+<a spec=fenced-frame>Send a beacon</a> with |destination map|, |pending event|'s [=pending event/eventType=],
+|pending event|'s [=pending event/eventData=], |pending event|'s [=pending event/attributionReportingEnabled=],
+and |pending event|'s [=pending event/contextOrigin=].
+
+Modify <a spec=fenced-frame>send a beacon</a> as follows:
+
+Add a [=boolean=] parameter called |attributionReportingEnabled| and an [=origin=]
+parameter called |contextOrigin|.
+
+After the step
+
+> Let <var ignore>destination url</var> be <var ignore>destination map</var>[|eventType|].
+
+add the steps
+
+1. Let |attributionReportingEligibility| be "<code>[=eligibility/unset=]</code>".
+1. If |attributionReportingEnabled| is true and the result of [=getting supported registrars=]
```

Comment:
Why do we need to check the registrars being empty here? Shouldn't that be handled universally in https://wicg.github.io/attribution-reporting-api/#set-attribution-reporting-headers ?

Reply:
Ideally we should handle this universally, but I don't think the implementation does this right now, therefore would prefer addressing this separately.

Also the checking here also allows setting process response handler properly when fetch is specified.

Reply:
> Also the checking here also allows setting process response handler properly when fetch is specified.

Could you elaborate on this? I'm not following.